### PR TITLE
silent warnings

### PR DESCRIPTION
**Pull request by mkortstiege**
.. just a few ones that popped up recently.

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by mkortstiege** [comment]
build error seems unrelated.

**Comment by janbar** [comment]
I fixed some warnings with #7488. But here I don't understand your logic because m_iTypeId is unsigned.

File: xbmc/pvr/PVRGUIInfo.cpp [hunk 1]
```diff
@@ -496,7 +496,7 @@ void CPVRGUIInfo::CharInfoBackendNumber(std::string &strValue) const
   size_t numBackends = m_backendProperties.size();
 
   if (numBackends > 0)
-    strValue = StringUtils::Format("%u %s %u", m_iCurrentActiveClient + 1, g_localizeStrings.Get(20163).c_str(), numBackends);
+    strValue = StringUtils::Format("%u %s %i", m_iCurrentActiveClient + 1, g_localizeStrings.Get(20163).c_str(), static_cast<int>(numBackends));
```
Comment (Paxxi) [inline xbmc/pvr/PVRGUIInfo.cpp]: What's this good for?

Comment (mkortstiege) [inline xbmc/pvr/PVRGUIInfo.cpp]: Outputting size_t as %u is causing a format warning with gcc.

Comment (xhaggi) [inline xbmc/pvr/PVRGUIInfo.cpp]: please use `"%" PRIuS`

**Comment by mkortstiege** [comment]
Updated and rebased.

**Comment by xhaggi** [comment]
thanks ;)

**Comment by mkortstiege** [comment]
jenkins build this please.

**Pull request merged**